rpc-server/src/modules/queries/methods.rs

```diff
@@ -76,10 +75,15 @@ async fn query_call(
         near_primitives::views::QueryRequest::ViewState {
             account_id,
             prefix,
-            include_proof: _,
+            include_proof,
         } => {
             crate::metrics::QUERY_VIEW_STATE_REQUESTS_TOTAL.inc();
-            view_state(&data, block, &account_id, prefix.as_ref()).await
+            if include_proof {
+                // We can calculate the proof for state only on archive nodes.
```

Comment:
This comment is not true. We can calculate proofs on regular nodes as well. What we cannot do is we cannot calculate it on ReadRPC.

Put a TODO here, please. Ideally, we would like to differentiate regular-data requests from the historical ones and proxy such requests to a corresponding NEAR RPC URL, but for now, let's proxy all such requests to the Archival node.